Automattic/wp-calypso
# Components: Remove unused `FormToggle` component

#### Changes proposed in this Pull Request

In #52915 we removed the last real need for a `FormToggle` component in Calypso. With that merged, `FormToggle` became a simple proxy of the `ToggleControl` component, and the only real difference is the way that the label is provided - in `FormToggle` we provide those as `children`, while in `ToggleControl` that's being done with the `label` prop. So `FormToggle` can just be removed in favor of `ToggleControl`, there is no real need to maintain that abstraction anymore.

This PR removes the unused Calypso `FormToggle` component completely in favor of `ToggleControl`.

This PR is split off #52917.

~This PR depends on (and contains) the following and needs to be rebased after they land:~

* [x] #52950
* [x] #52951
* [x] #52952
* [x] #52953
* [x] #52954
* [x] #52956

#### Testing instructions

* Verify Calypso builds and all tests pass.
* Verify the `FormToggle` component is no longer used anywhere.
